boa/src/builtins/object/mod.rs

```diff
@@ -481,4 +493,44 @@ impl Object {
             Value::from(own_prop.enumerable())
         }))
     }
+
+    /// `Object.assign( target, ...sources )`
+    ///
+    /// This method copies all enumerable own properties from one or more
+    /// source objects to a target object. It returns the target object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-object.assign
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
+    pub fn assign(_: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let mut to = args
+            .get(0)
+            .cloned()
+            .unwrap_or_default()
+            .to_object(context)?;
+
+        if args.len() == 1 {
```

Comment:
I would change this to:
```suggestion
        if args.len() <= 1 {
```

Since in the case that no argument is provided, the `to` variable will be `undefined` (which is correct), but this would be `false`, which could cause unintended behaviour.

Reply:
Actually, never mind, the `to_object()` above fails with a `TypeError`, which is the expected behaviour.